great_expectations_provider/operators/great_expectations.py

```diff
@@ -44,121 +78,404 @@ class GreatExpectationsOperator(BaseOperator):
+            )
+
+        # A data asset name is also used to determine if a runtime env will be used; if it is not passed in,
+        # then the data asset name is assumed to be configured in the data context passed in.
+        if (
+            self.is_dataframe or self.query_to_validate or self.conn_id
+        ) and not self.data_asset_name:
+            raise ValueError(
+                "A data_asset_name must be specified with a runtime_data_source or conn_id."
+            )
+
+        # Check that at most one of the arguments is passed to set a checkpoint
+        if self.checkpoint_name and self.checkpoint_config:
+            raise ValueError(
+                "Exactly one, or neither, of checkpoint_name or checkpoint_config may be specified. If neither is"
+                " specified, the default Checkpoint is used."
+            )
+
+        if (
+            not (self.checkpoint_name or self.checkpoint_config)
+            and not self.expectation_suite_name
+        ):
+            raise ValueError(
+                "An expectation_suite_name must be supplied if neither checkpoint_name nor checkpoint_config are."
+            )
+
+        if type(self.checkpoint_config) == CheckpointConfig:
+            self.checkpoint_config = self.checkpoint_config.to_json_dict()
+            self.checkpoint_config = {
+                k: v for k, v in self.checkpoint_config.items() if v
+            }
+
+    def build_configured_sql_datasource_config_from_conn_id(
+        self,
+    ) -> Datasource:
+        datasource_config = {
+            "name": f"{self.conn.conn_id}_configured_datasource",
+            "execution_engine": {
+                "module_name": "great_expectations.execution_engine",
+                "class_name": "SqlAlchemyExecutionEngine",
+                "connection_string": self.conn.get_uri(),
+            },
+            "data_connectors": {
+                "default_configured_asset_sql_data_connector": {
+                    "module_name": "great_expectations.datasource.data_connector",
+                    "class_name": "ConfiguredAssetSqlDataConnector",
+                    "assets": {
+                        f"{self.data_asset_name}": {
+                            "module_name": "great_expectations.datasource.data_connector.asset",
+                            "class_name": "Asset",
+                            "schema_name": f"{self.conn.schema}",
+                            "batch_identifiers": ["airflow_run_id"],
+                        },
+                    },
+                },
+            },
+        }
+        return Datasource(**datasource_config)
+
+    def build_configured_sql_datasource_batch_request(self):
+        batch_request = {
+            "datasource_name": f"{self.conn.conn_id}_configured_sql_datasource",
+            "data_connector_name": "default_configured_asset_sql_data_connector",
+            "data_asset_name": f"{self.data_asset_name}",
```

Comment:
Not sure if applicable, but `BatchRequest` can admit a number of additional parameters, aimed at narrowing down and/or selecting the `Batch` objects returned.  This `build_` method only uses the required triumvirate.  Thanks.

Reply:
Yep, this is just for a default in case a more configured one is not passed in.

Reply:
@denimalpaca So, just to confirm, if a more detailed configuration is passed in, will it be accepted?  If so, could you please point me to the code segment that accomplishes this?  Thanks!

Reply:
If I'm understanding how checkpoints are built correctly, the [`checkpoint_kwargs`](https://github.com/astronomer/airflow-provider-great-expectations/pull/57/files/739c86a35f4feae1b6f3196b4407975171c6d2da#diff-b38c90d08ff1bba880fae02c122f7b317802277d847d11f11e0df3f55656a5c2R102) argument can be passed into the operator, and if so they can get used to run the checkpoint [here](https://github.com/astronomer/airflow-provider-great-expectations/pull/57/files/739c86a35f4feae1b6f3196b4407975171c6d2da#diff-b38c90d08ff1bba880fae02c122f7b317802277d847d11f11e0df3f55656a5c2R467).

Although at the time of writing, this code may never get run if the default data context is used because the default batch request will also always be built, meaning the `checkpoint_kwargs` logic will never be run. I can fix this part, though.

Reply:
Seconding @denimalpaca. Also, I think what you're saying is right, and we should fix it.

@alexsherstinsky, the user can use an existing Checkpoint with a detailed BatchRequest, or pass in a detailed BatchRequest to `checkpoint_kwargs`.

Reply:
Fixed :)